src/pkg/artifact/model.go

```diff
@@ -48,6 +48,15 @@ type Artifact struct {
 	References        []*Reference           `json:"references"` // child artifacts referenced by the parent artifact if the artifact is an index
 }
 
+// ResolveArtifactType returns the artifact type of the artifact, prefer ArtifactType, use MediaType if ArtifactType is empty.
+func (a *Artifact) ResolveArtifactType() string {
```

Comment:
I am pretty-sure that this change is not necessary and can be avoided to reduce scope of change to only adding artifact processor, but let's see what code-owners have to say.

Reply:
The logic of this method is correct, and wrapping it into a shared method would reduce duplication. Works for me.